contracts/IdentityFactory.sol

```diff
@@ -8,26 +8,59 @@ contract IdentityFactory is Initializable {
 
   event IdentityCreated(address indexed identity);
 
+  /**
+  * @dev Keep track or identities created with the factory contract
+  * This is necessary because there is no way to tell on chain
+  * if the identity interface is implemented in a certain way and
+  * the Centrifuge Protocol requires on chain key history. A key
+  * should be revoked and not removed
+  */
+  mapping(address => bool) internal _identities;
+
   /**
   * Deploys a new identity and transfers the ownership to the sender
   */
   function createIdentity()
   external
   {
-    Identity identity_ = new Identity(msg.sender);
-    emit IdentityCreated(address(identity_));
+
```

Comment:
Why not just call `createIdentityFor(msg.sender,....)` here?

Reply:
you could , the only downside would be that createIdentityFor will be public